src/Implementations/BaseMessageBus.ts

```diff
@@ -0,0 +1,107 @@
+import {HandlersMap} from '../Lib/HandlersMap';
+import {Message} from '../Lib/Message';
+import {MessageBus} from '../Lib/MessageBus';
+import {MessageBusMiddleware} from '../Lib/MessageBusMiddleware';
+import {MessageHandler} from '../Lib/MessageHandler';
+import {MessageHandlerObject} from '../Lib/MessageHandlerObject';
+import {isMessage} from './TypeCheckers/isMessage';
+import {isMessageHandler} from './TypeCheckers/isMessageHandler';
+import {isMessageHandlerFunction} from './TypeCheckers/isMessageHandlerFunction';
+import {isMessageHandlerObject} from './TypeCheckers/isMessageHandlerObject';
+
+export abstract class BaseMessageBus implements MessageBus {
+    protected handlersMap: HandlersMap = {};
+    protected middlewareList: MessageBusMiddleware<any>[] = [];
```

Comment:
Middlewares should match some interface. It can be a single functions where message is passed or classes with ex. handle function which accepts message as a parameter. For now if middleware class will not have a handle method `handle` will throw an error about calling undefined function.